src/cpp/dataset.cpp

```diff
@@ -31,54 +31,68 @@
 
 using namespace SmartRedis;
 
-DataSet::DataSet(const std::string& name)
+// DataSet constructor
+DataSet::DataSet(const std::string& _name)
+ : name(_name)
 {
-    this->name = name;
+    // NOP
 }
 
+// Add a tensor to the DataSet.
 void DataSet::add_tensor(const std::string& name,
                          void* data,
                          const std::vector<size_t>& dims,
                          const TensorType type,
                          MemoryLayout mem_layout)
 {
-    this->_add_to_tensorpack(name, data, dims,
+    _add_to_tensorpack(name, data, dims,
                              type, mem_layout);
-    this->_metadata.add_string(".tensor_names", name);
-    return;
+    _metadata.add_string(".tensor_names", name);
 }
 
+// Add metadata scalar field (non-string) with value to the DataSet.
+// If the field does not exist, it will be created. If the field exists,
+// the value will be appended to existing field.
 void DataSet::add_meta_scalar(const std::string& name,
                               const void* data,
                               const MetaDataType type)
 {
-    this->_metadata.add_scalar(name, data, type);
-    return;
+    _metadata.add_scalar(name, data, type);
 }
 
+// Add metadata string field with value to the DataSet. If the field
+// does not exist, it will be created. If the field exists the value
+// will be appended to existing field.
 void DataSet::add_meta_string(const std::string& name,
                               const std::string& data)
 {
-    this->_metadata.add_string(name, data);
-    return;
+    _metadata.add_string(name, data);
 }
 
+// Get the tensor data, dimensions, and type for the tensor in the DataSet.
+// This function will allocate and retain management of the memory for
+// the tensor data.
 void DataSet::get_tensor(const std::string& name,
                          void*& data,
                          std::vector<size_t>& dims,
                          TensorType& type,
                          MemoryLayout mem_layout)
 {
-    this->_enforce_tensor_exists(name);
+    _enforce_tensor_exists(name);
     // Clone the tensor in the DataSet
-    TensorBase* tensor = this->_get_tensorbase_obj(name);
-    this->_tensor_memory.add_tensor(tensor);
+    TensorBase* tensor = _get_tensorbase_obj(name);
+    if (tensor == NULL) {
```

Comment:
Is there a code path to hit this?

Reply:
Again, never know when an error or other surprise can come up as code changes over time, so best practice is to code defensively